lib/cli.js

```diff
@@ -17,86 +17,82 @@
 'use strict';
 
 var fs = require('fs');
+var path = require('path');
 
 var getGitFolder = require('./getGitFolder');
 var validateMessage = require('../index');
 
-// hacky start if not run by mocha :-D
-// istanbul ignore next
-if (process.argv.join('').indexOf('mocha') === -1) {
-  var bufferToString = function (buffer) {
-    var hasToString = buffer && typeof buffer.toString === 'function';
-    return hasToString && buffer.toString();
-  };
+var commitMsgFileOrText = process.argv[2];
+var commitErrorLogPath = process.argv[3];
 
-  var getFileContent = function (filePath) {
-    try {
-      var buffer = fs.readFileSync(filePath);
-      return bufferToString(buffer);
-    } catch (err) {
-      // Ignore these error types because is most likely it is validating
-      // a commit from a text instead of a file
-      if(err && err.code !== 'ENOENT' && err.code !== 'ENAMETOOLONG') {
-        throw err;
-      }
-    }
-  };
+// On running the validation over a text instead of git files such as COMMIT_EDITMSG and GITGUI_EDITMSG
+// is possible to be doing that the from anywhere. Therefore the git directory might not be available.
+var gitDirectory;
+try {
+  gitDirectory = getGitFolder();
+
+  if (!commitErrorLogPath) {
+    commitErrorLogPath = path.resolve(gitDirectory, 'logs/incorrect-commit-msgs');
+  }
+} catch (err) {}
+
+var bufferToString = function (buffer) {
+  var hasToString = buffer && typeof buffer.toString === 'function';
+
+  return hasToString && buffer.toString();
+}
+
+var getCommit = function (msgFileOrText) {
+  var commit = getCommitFromFile(msgFileOrText);
+
+  if (commit === null && !msgFileOrText) {
```

Comment:
Because at this point we have no idea if the parameter is a file or a text. Hence its name.
So we start checking if it's a valid file.
If not, it is a text.
In case this parameter is not provided, we fall back to COMMIT_EDITMSG.